packages/pluggableWidgets/datagrid-date-filter-web/src/DatagridDateFilter.tsx

```diff
@@ -154,38 +185,52 @@ function getAttributeTypeErrorMessage(type?: string): string | null {
 function getFilterCondition(
     listAttribute: ListAttributeValue,
     value: Date | null,
+    rangeValues: [Date | null, Date | null] | null,
     type: DefaultFilterEnum
 ): FilterCondition | undefined {
-    if (!listAttribute || !listAttribute.filterable || !value) {
+    if (
+        !listAttribute ||
+        !listAttribute.filterable ||
+        (type !== "between" && !value) ||
+        (type === "between" && (!rangeValues || !rangeValues[0] || !rangeValues[1]))
+    ) {
         return undefined;
     }
 
     const filterAttribute = attribute(listAttribute.id);
-    const dateValue = changeTimeToMidnight(value);
+    const dateValue = value && changeTimeToMidnight(value);
     switch (type) {
+        case "between":
+            // >= Start day at midnight and <= End day at 23:59:59:999
+            const startDate = changeTimeToMidnight(rangeValues![0]!);
+            const endDate = changeTimeToMidnight(rangeValues![1]!);
+            return and(
+                greaterThanOrEqual(filterAttribute, literal(startDate)),
+                lessThanOrEqual(filterAttribute, literal(new Date(addDays(endDate, 1).getTime() - 1)))
+            );
         case "greater":
             // > Day +1 at midnight -1ms
-            return greaterThan(filterAttribute, literal(new Date(addDays(dateValue, 1).getTime() - 1)));
+            return greaterThan(filterAttribute, literal(new Date(addDays(dateValue!, 1).getTime() - 1)));
```

Comment:
These casts look so bad but I'm also not sure how to get rid of them 😓  I would almost think that handling the between case outside this switch statement in its own if statement would make the code way cleaner (cuz then you also don't have to case `rangeValues` 😅  You don't _have_ to change it tho